test/jdk/java/lang/ClassLoader/loadLibraryUnload/LoadLibraryUnload.java

```diff
@@ -0,0 +1,159 @@
+                Class clazz = findLoadedClass(name);
+                if (clazz == null) {
+                    try {
+                        clazz = findClass(name);
+                    } catch (ClassNotFoundException ignore) {
+                    }
+                    if (clazz == null) {
+                        clazz = super.loadClass(name);
+                    }
+                }
+                return clazz;
+            }
+        }
+    }
+
+    private static class LoadLibraryFromClass implements Runnable {
+        Object object;
+        Method method;
+
+        public LoadLibraryFromClass(Class<?> fromClass) {
+            try {
+                this.object = fromClass.newInstance();
+                this.method = fromClass.getDeclaredMethod("loadLibrary");
+            } catch (Exception error) {
+                throw new Error(error);
+            }
+        }
+
+        @Override
+        public void run() {
+            try {
+                method.invoke(object);
+            } catch (Exception error) {
+                throw new Error(error);
+            }
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        Class<?> clazz = null;
+        List<Thread> threads = new ArrayList<>();
+
+        for (int i = 0 ; i < 5 ; i++) {
+            // 5 loaders and 10 threads in total.
+            // winner loads the library in 2 threads
+            clazz = new TestLoader().loadClass("p.Class1");
+            threads.add(new Thread(new LoadLibraryFromClass(clazz)));
+            threads.add(new Thread(new LoadLibraryFromClass(clazz)));
+        }
+
+        final Set<Throwable> exceptions = ConcurrentHashMap.newKeySet();
+        threads.forEach( t -> {
+            t.setUncaughtExceptionHandler((th, ex) -> {
+                // collect the root cause of each failure
+                Throwable rootCause = ex;
+                while((ex = ex.getCause()) != null) {
+                    rootCause = ex;
+                }
+                exceptions.add(rootCause);
+            });
+            t.start();
+        });
+
+        // wait for all threads to finish
+        for (Thread t : threads) {
+            t.join();
+        }
+
+        // expect all errors to be UnsatisfiedLinkError
+        boolean allAreUnsatisfiedLinkError = exceptions
+                .stream()
+                .map(e -> e instanceof UnsatisfiedLinkError)
+                .reduce(true, (i, a) -> i && a);
+
+        // expect exactly 8 errors
+        Asserts.assertTrue(exceptions.size() == 8,
+                "Expected to see 8 failing threads");
+
+        Asserts.assertTrue(allAreUnsatisfiedLinkError,
+                "All errors have to be UnsatisfiedLinkError");
+
+        WeakReference<Class> wClass = new WeakReference<>(clazz);
+
+        // release strong refs
+        clazz = null;
+        threads = null;
+        exceptions.clear();
+        waitForUnload(wClass);
```

Comment:
`jdk.test.lib.util.ForceGC` can be used here.  You can reference test/jdk/java/lang/ClassLoader/nativeLIbrary/NativeLibraryTest.java.